src/core/iomgr/tcp_server.h

```diff
@@ -80,4 +80,8 @@ int grpc_tcp_listener_get_port(grpc_tcp_listener *listener);
 void grpc_tcp_listener_ref(grpc_tcp_listener *listener);
 void grpc_tcp_listener_unref(grpc_tcp_listener *listener);
 
+/* potentially (trivially) implementable: */
+int grpc_tcp_listener_get_fd(grpc_tcp_listener *listener, unsigned index);
```

Comment:
The only use I've seen of grpc_tcp_listener in existing code is to get the port. You could have add_port() return the port, and keep listener completely private. For the new stuff, you'd go with my earlier idea of passing the grpc_tcp_server to on_accept_cb with an index and fixing grpc_tcp_server_get_fd().

That main thing to do is clarify whether grpc_tcp_listener is private or not. It's sort of stuck in between right now, not completely making sense either way.

I could send a pull request showing what this would look like.